pkg/distributor/ha_tracker.go

```diff
@@ -242,18 +242,20 @@ func replicasNotMatchError(replica, elected string) error {
 	return httpgrpc.Errorf(http.StatusAccepted, "replicas did not mach, rejecting sample: replica=%s, elected=%s", replica, elected)
 }
 
-// Modifies the labels parameter in place, removing labels that match
-// the replica or cluster label and returning their values. Returns an error
-// if we find one but not both of the labels.
+// Searches for the values of replicaLabel and clusterLabel
+// returning their values. Returns an error if we find one but not both of the labels.
 func findHALabels(replicaLabel, clusterLabel string, labels []client.LabelAdapter) (string, string) {
 	var cluster, replica string
 	var pair client.LabelAdapter
 
-	for _, pair = range labels {
+	for i := 0; i < len(labels) && replica == "" && cluster == ""; i++ {
```

Comment:
I would suggest to cover the new case the `clusterLabel` is `""` in `TestFindHALabels` (`ha_tracker_test.go`)

Reply:
fixed the if

Reply:
dne